.github/workflows/mac.yml

```diff
@@ -105,7 +105,7 @@ jobs:
           --use_xnnpack \
           --use_binskim_compliant_compile_flags \
           --ios \
-          --apple_deploy_target=13.0 \
+          --apple_deploy_target=15.1 \
```

Comment:
How are we choosing minor versions? e.g. why 15.1 vs 15.0 or 15.8?

Reply:
The latest React Native requires iOS 15.1+.  See #23292